cmd/celestia/blob.go

```diff
@@ -0,0 +1,200 @@
+	Use:   "blob [command]",
+	Short: "Allows to interact with the Blob Service via JSON-RPC",
+	Args:  cobra.NoArgs,
+}
+
+var getCmd = &cobra.Command{
+	Use:   "get [params]",
+	Args:  cobra.ExactArgs(3),
+	Short: "Returns the blob for the given namespace by commitment at a particular height.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := rpcClient(cmd.Context())
+		if err != nil {
+			return err
+		}
+
+		num, err := strconv.ParseUint(args[0], 10, 64)
+		if err != nil {
+			return fmt.Errorf("error parsing a height:%v", err)
+		}
+
+		namespace, err := parseV0Namespace(args[1])
+		if err != nil {
+			return fmt.Errorf("error parsing a namespace:%v", err)
+		}
+
+		commitment, err := base64.StdEncoding.DecodeString(args[2])
+		if err != nil {
+			return fmt.Errorf("error parsing a commitment:%v", err)
+		}
+
+		var output []byte
+		blob, err := client.Blob.Get(cmd.Context(), num, namespace, commitment)
+		if err != nil {
+			output, err = prepareOutput(err)
+			if err != nil {
+				return err
+			}
+		} else {
+			output, err = prepareOutput(blob)
+			if err != nil {
+				return err
+			}
+		}
+
+		fmt.Fprintln(os.Stdout, string(output))
+		return nil
+	},
+}
+
+var getAllCmd = &cobra.Command{
+	Use:   "getAll [params]",
+	Args:  cobra.ExactArgs(2),
+	Short: "Returns all blobs for the given namespace at a particular height.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := rpcClient(cmd.Context())
+		if err != nil {
+			return err
+		}
+
+		height, err := strconv.ParseUint(args[0], 10, 64)
+		if err != nil {
+			return fmt.Errorf("error parsing a height:%v", err)
+		}
+
+		namespace, err := parseV0Namespace(args[1])
+		if err != nil {
+			return fmt.Errorf("error parsing a namespace:%v", err)
+		}
+
+		var output []byte
+		blobs, err := client.Blob.GetAll(cmd.Context(), height, []share.Namespace{namespace})
+		if err != nil {
+			output, err = prepareOutput(err)
+			if err != nil {
+				return err
+			}
+		} else {
+			output, err = prepareOutput(blobs)
+			if err != nil {
+				return err
+			}
+		}
+
+		fmt.Fprintln(os.Stdout, string(output))
+		return nil
+	},
+}
+
+var submitCmd = &cobra.Command{
+	Use:   "submit [params]",
+	Args:  cobra.ExactArgs(2),
+	Short: "Submit the blob at the given namespace. Note: only one blob is allowed to submit through the RPC.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := rpcClient(cmd.Context())
+		if err != nil {
+			return err
+		}
+
+		namespace, err := parseV0Namespace(args[0])
+		if err != nil {
+			return fmt.Errorf("error parsing a namespace:%v", err)
+		}
+
+		parsedBlob, err := blob.NewBlobV0(namespace, []byte(args[1]))
+		if err != nil {
+			return fmt.Errorf("error creating a blob:%v", err)
+		}
+
+		var output []byte
+		height, err := client.Blob.Submit(cmd.Context(), []*blob.Blob{parsedBlob})
+		if err != nil {
+			output, err = prepareOutput(err)
+			if err != nil {
+				return err
+			}
```

Comment:
Why do we need to give error back in json and not just a string?

Reply:
Because in some cases IMO we have to emphasise that the error is a result of request, received from the service. Otherwise you will have printed error and recommendations for usage current command, that does not make sense.